### PR TITLE
chore: move some file functions to upload service

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
## Proposed changes (including videos or screenshots)

FED-114
I'm moving those functions to this service because the Federation module depends on those functions, and since we are moving the module to an external service, we need to enable the dependencies via service methods.

### CODE REVIEW

File: apps/meteor/server/services/upload/service.ts [hunk 1]
```diff
@@ -22,12 +22,31 @@ export class UploadService extends ServiceClassInternal implements IUploadServic
...
+		roomId: string,
+		user: IUser,
+	): Promise<{
+		files: {
+			_id: string;
+			name: string | undefined;
+			type: string | undefined;
+		}[];
+		attachments: MessageAttachment[];
+	}> {
```
Comment (KevLehman) [inline apps/meteor/server/services/upload/service.ts:49]: Not sure what's better, typing this again, or type it as `ReturnType<parseFileIntoMessageAttachments>` now that's properly typed

File: apps/meteor/server/services/federation/infrastructure/rocket-chat/adapters/File.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export class RocketChatFileAdapter {
 			try {
 				const { files, attachments } = await parseFileIntoMessageAttachments(uploadedFile, internalRoomId, internalUser);
 
-				resolve({ files, attachments });
+				resolve({ files: files as FileProp[], attachments });
```
Comment (KevLehman) [inline apps/meteor/server/services/federation/infrastructure/rocket-chat/adapters/File.ts:26]: If you're casting files as FileProp, why not making `parseFileINtoMessageAttachment` return a `FileProp[]`?

**Review by KevLehman — CHANGES_REQUESTED** [review]

File: apps/meteor/server/services/federation/infrastructure/rocket-chat/adapters/File.ts (hunk 2 above)
Comment (MarcosSpessatto) [reply apps/meteor/server/services/federation/infrastructure/rocket-chat/adapters/File.ts:26]: makes sense 🤙

File: packages/core-typings/src/IMessage/MessageAttachment/Files/FileProp.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 export type FileProp = {
 	_id: string;
-	name: string;
-	type: string;
-	format: string;
-	size: number;
+	name?: string;
+	type?: string;
+	format?: string;
+	size?: number;
```
Comment (KevLehman) [inline packages/core-typings/src/IMessage/MessageAttachment/Files/FileProp.ts:6]: Hmmm, a type with only ID as required, maybe it's not the proper type for your use case?

**Review by KevLehman — APPROVED** [review]

**Review by sampaiodiego — DISMISSED** [review]

**Review by KevLehman — APPROVED** [review]

**Review by elfajriharsuda — APPROVED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**